test/e2e/framework/kcp-ingress.go

```diff
@@ -0,0 +1,284 @@
+	c.t.Logf("running: %v", strings.Join(cmd.Args, " "))
+	logFile, err := os.Create(filepath.Join(c.artifactDir, "kcp.log"))
+	if err != nil {
+		cleanupCancel()
+		return fmt.Errorf("could not create log file: %w", err)
+	}
+	log := bytes.Buffer{}
+	writers := []io.Writer{&log, logFile}
+	mw := io.MultiWriter(writers...)
+	cmd.Stdout = mw
+	cmd.Stderr = mw
+	if err := cmd.Start(); err != nil {
+		cleanupCancel()
+		return err
+	}
+	go func() {
+		defer func() { cleanupCancel() }()
+		err := cmd.Wait()
+		data := c.filterKcpIngressLogs(&log)
+		if err != nil && ctx.Err() == nil {
+			// we care about errors in the process that did not result from the
+			// context expiring and us ending the process
+			c.t.Errorf("`kcp` failed: %w logs:\n%v", err, data)
+		}
+	}()
+	return nil
+}
+
+func (c *kcpIngress) filterKcpIngressLogs(logs *bytes.Buffer) string {
+	output := strings.Builder{}
+	scanner := bufio.NewScanner(logs)
+	for scanner.Scan() {
+		line := scanner.Bytes()
+		ignored := false
+		for _, ignore := range [][]byte{
+			// TODO: a rebase of our k8s fork on something newer will remove the following messages
+			[]byte(`flowcontrol.apiserver.k8s.io/v1beta1 FlowSchema is deprecated in v1.23+, unavailable in v1.26+`),
+			[]byte(`flowcontrol.apiserver.k8s.io/v1beta1 PriorityLevelConfiguration is deprecated in v1.23+, unavailable in v1.26+`),
+			// TODO: some careful thought on context cancellation might fix the following error
+			[]byte(`clientconn.go:1326] [core] grpc: addrConn.createTransport failed to connect to`),
+		} {
+			if bytes.Contains(line, ignore) {
+				ignored = true
+				continue
+			}
+		}
+		if ignored {
+			continue
+		}
+		_, err := output.Write(append(line, []byte(`\n`)...))
+		if err != nil {
+			c.t.Logf("failed to write log line: %v", err)
+		}
+	}
+	return output.String()
+}
+
+// Config exposes a copy of the client config for this server.
+func (c *kcpIngress) Config() *rest.Config {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	if c.cfg == nil {
+		return nil
+	}
+	return rest.CopyConfig(c.cfg)
+}
+
+// Ready blocks until the server is healthy and ready. Before returning,
+// goroutines are started to ensure that the test is failed if the server
+// does not remain so.
+func (c *kcpIngress) Ready() error {
+	if err := c.loadCfg(); err != nil {
+		return err
+	}
+	if c.ctx.Err() != nil {
+		// cancelling the context will preempt derivative calls but not this
+		// main Ready() body, so we check before continuing that we are live
+		return fmt.Errorf("failed to wait for readiness: %w", c.ctx.Err())
+	}
+	cfg := c.Config()
+	if cfg.NegotiatedSerializer == nil {
+		cfg.NegotiatedSerializer = scheme.Codecs.WithoutConversion()
+	}
+	client, err := rest.UnversionedRESTClientFor(cfg)
+	if err != nil {
+		return fmt.Errorf("failed to create unversioned client: %w", err)
+	}
+
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	for _, endpoint := range []string{"/livez", "/readyz"} {
```

Comment:
Does kcp-ingress serve these?

Reply:
No, I will have to modify the controller to provide some health/live endpoints that could be used by the e2e framework.